extendr-api/src/robj/into_robj.rs

```diff
@@ -721,28 +704,49 @@ macro_rules! impl_from_as_iterator {
 //     }
 // } //
 
-// Template constants are still unstable in rust.
-impl_from_iter! {[T; 1]}
-impl_from_iter! {[T; 2]}
-impl_from_iter! {[T; 3]}
-impl_from_iter! {[T; 4]}
-impl_from_iter! {[T; 5]}
-impl_from_iter! {[T; 6]}
-impl_from_iter! {[T; 7]}
-impl_from_iter! {[T; 8]}
-impl_from_iter! {[T; 9]}
-impl_from_iter! {[T; 10]}
-impl_from_iter! {[T; 11]}
-impl_from_iter! {[T; 12]}
-impl_from_iter! {[T; 13]}
-impl_from_iter! {[T; 14]}
-impl_from_iter! {[T; 15]}
-impl_from_iter! {[T; 16]}
-impl_from_iter! {[T; 17]}
-impl_from_iter! {[T; 18]}
-impl_from_iter! {[T; 19]}
-impl_from_iter! {Vec<T>}
-impl_from_iter! {&Vec<T>}
+impl<'a, T, const N: usize> From<[T; N]> for Robj
+where
+    Self: 'a,
+    T: ToVectorValue,
+{
+    fn from(val: [T; N]) -> Self {
+        fixed_size_collect(val.into_iter(), N)
+    }
+}
+
+impl<'a, T, const N: usize> From<&'a [T; N]> for Robj
+where
+    Self: 'a,
+    &'a T: ToVectorValue + 'a,
+{
+    fn from(val: &'a [T; N]) -> Self {
+        fixed_size_collect(val.into_iter(), N)
+    }
+}
+
+impl<'a, T, const N: usize> From<&'a mut [T; N]> for Robj
+where
+    Self: 'a,
+    &'a mut T: ToVectorValue + 'a,
+{
+    fn from(val: &'a mut [T; N]) -> Self {
+        fixed_size_collect(val.into_iter(), N)
+    }
+}
+
+impl<T: ToVectorValue + Clone> From<&Vec<T>> for Robj {
+    fn from(value: &Vec<T>) -> Self {
+        let len = value.len();
+        fixed_size_collect(value.iter().cloned(), len)
+    }
+}
+
+impl<T: ToVectorValue> From<Vec<T>> for Robj {
+    fn from(value: Vec<T>) -> Self {
+        let len = value.len();
+        fixed_size_collect(value.into_iter(), len)
+    }
+}
 
 impl_from_into_iter! {&'a [T]}
```

Comment:
If this is the only use of the macro now, should it be kept here, or maybe we can drop the macro in favor of just writing it out as is?

Reply:
I did make a PR where I tried to get rid of everything. Frankly, I'm just hoping for hyper scoped PRs, that get merged, and then I'll do whatever. But I'll take a look at this now.

Reply:
Oh this can be inlined as you say.. Oops.. I got rid of it now.